server/parsec/components/postgresql/migrations/0001_initial.sql

```diff
@@ -302,22 +314,24 @@ CREATE TABLE realm_user_role (
     realm INTEGER REFERENCES realm (_id) NOT NULL,
     user_ INTEGER REFERENCES user_ (_id) NOT NULL,
     -- NULL if access revocation
-    role realm_role,
+    role REALM_ROLE,
     certificate BYTEA NOT NULL,
-    certified_by INTEGER REFERENCES device(_id) NOT NULL,
+    certified_by INTEGER REFERENCES device (_id) NOT NULL,
     certified_on TIMESTAMPTZ NOT NULL
 );
 
-CREATE TYPE realm_archiving_configuration AS ENUM ('AVAILABLE', 'ARCHIVED', 'DELETION_PLANNED');
+CREATE TYPE realm_archiving_configuration AS ENUM (
```

Comment:
The line `330` change the name of the type to UPPERCASE, should it be the same here ?

Reply:
Good catch. The docs show some configuration options to fine-tune capitalisation rules. I'll check that.
 
```cfg
# The default configuration for capitalisation rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
# Typically we find users rely on syntax highlighting rather than
# capitalisation to distinguish between keywords and identifiers.
# Clearly, if your organisation has already settled on uppercase
# formatting for any of these syntax elements then set them to "upper".
# See https://stackoverflow.com/questions/608196/why-should-i-capitalize-my-sql-keywords-is-there-a-good-reason
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
```

https://docs.sqlfluff.com/en/stable/configuration.html#new-project-configuration

Reply:
So after some investigation, they are not considered the same "thing" in SQL. 

For example, here `sequester_service_type` is considered an _identifier_ and is expected to be lower case (as all other table or column identifiers, such as `realm` or `user_id`):
```sql
CREATE TYPE sequester_service_type AS ENUM ('STORAGE', 'WEBHOOK');
````

Here, `SEQUESTER_SERVICE_TYPE` is considered a _data type_ and is expected to be upper case (as all other data types such as `INTEGER` or `TIMESTAMPTZ`)
```sql
CREATE TABLE sequester_service (
    -- ...
    service_type SEQUESTER_SERVICE_TYPE NOT NULL,
````

I've just added rules to .sqlfluff files to make this explicit.

I guess this answer applies to all other comments.

Reply:
TIL, yes that apply for the other comment :+1: